src/hwdb/hwdb.c

```diff
@@ -340,12 +340,13 @@ static int64_t trie_store_nodes(struct trie_f *trie, struct trie_node *node) {
                 int64_t child_off;
 
                 child_off = trie_store_nodes(trie, node->children[i].child);
-                if (child_off < 0) {
-                        free(children);
+                if (child_off < 0)
                         return child_off;
-                }
-                children[i].c = node->children[i].c;
-                children[i].child_off = htole64(child_off);
+
+                children[i] = (struct trie_child_entry_f) {
+                        .c = node->children[i].c,
+                        .child_off = htole64(child_off),
+                };
```

Comment:
Yeah, this sets the whole structure (there's a padding field which we don't use but it should be initialized so that garbage is not written to the file), so we avoid the need to zero the memory beforehand.

It also matches a similar branch in the next loop, I think it looks nicer this way.